### PR TITLE
Use Scripts instead of Grunt

### PR DESCRIPTION
**Pull request by aminya**
### Description of the change
This uses simple npm scripts instead of Grunt. This is a much better approach used in big projects such as [Github itself](https://github.com/desktop/desktop/blob/6c62f87e942723704f7c21192266b8b91418bedb/package.json#L8). 

### Verification
The CI and tests all pass. This does not change any functionality.

### Release Notes
N/A

### CODE REVIEW

**Comment by aminya** [comment]
Upstreamed https://github.com/atom/apm/pull/903

File: package.json [hunk 1]
```diff
@@ -15,11 +15,14 @@
     "apm": "./bin/apm"
   },
   "scripts": {
-    "build": "node script/check-version.js && grunt",
-    "prepare": "grunt prepare",
+    "check-version": "node script/check-version.js",
+    "clean": "shx rm -rf lib bin/node_darwin_x64 && shx mkdir -p lib",
```
Comment (DeeDeeG) [inline package.json:19]: I think the old Grunt "clean" task was really out-of-date. (This "node_darwin_x64" thing was added 7 years ago https://github.com/atom/apm/blame/v2.5.0/Gruntfile.coffee#L34). We never end up with anything called `bin/node_darwin_x64` anymore.

It's just `bin/node` now, on both Linux and macOS. And on Windows it's `bin/node.exe`.

File: package.json [hunk 2]
```diff
@@ -15,11 +15,14 @@
...
+    "lint": "coffeelint src spec",
+    "coffee": "coffee --compile --output lib src",
+    "build": "npm run clean && npm run coffee",
```
Comment (DeeDeeG) [inline package.json:22]: The original `build` script did this stuff but also ran the linter at the end.

I think this should be `npm run clean && npm run coffee && npm run lint`

File: package.json [hunk 3]
```diff
@@ -15,11 +15,14 @@
     "apm": "./bin/apm"
   },
   "scripts": {
-    "build": "node script/check-version.js && grunt",
-    "prepare": "grunt prepare",
+    "check-version": "node script/check-version.js",
+    "clean": "shx rm -rf lib bin/node_darwin_x64 && shx mkdir -p lib",
+    "lint": "coffeelint src spec",
+    "coffee": "coffee --compile --output lib src",
```
Comment (DeeDeeG) [inline package.json:21]: This caused lines like this to be added at the top of every file in `lib/`:

`// Generated by CoffeeScript 2.5.1`

The old `coffee` task was equivalent to running `coffee --no-header --compile --output lib src `

I would recommend adding `--no-header` again.

**Comment by DeeDeeG** [comment]
This is a really good idea overall. Thanks for doing this. :+1:

File: package.json (hunk 2 above)
Comment (aminya) [reply package.json:22]: I don't want to follow what was being used blindly. I don't see a reason to lint the code after coffeescript has compiled the code.

File: package.json (hunk 1 above)
Comment (aminya) [reply package.json:19]: This is already fixed on master.

Comment (DeeDeeG) [reply package.json:19]: I noticed that, thanks. But master still "deletes" `node_darwin_x64` which is never going to exist.

File: package.json (hunk 2 above)
Comment (DeeDeeG) [reply package.json:22]: Okay, that sounds basically reasonable.

But doesn't this mean we don't run the linter on pull requests/in CI anymore? I would still want to do that to make sure our code is formatted cleanly.

Comment (aminya) [reply package.json:22]: Linting is part of test script.

Comment (DeeDeeG) [reply package.json:22]: I didn't see that, thanks for pointing it out.

**Pull request merged**